src/GameLogic/PlayerActions/ItemConsumeActions/HigherRefineStoneConsumeHandlerPlugIn.cs

```diff
@@ -9,17 +9,17 @@ namespace MUnique.OpenMU.GameLogic.PlayerActions.ItemConsumeActions;
 using MUnique.OpenMU.PlugIns;
 
 /// <summary>
-/// Consume handler for the Lower Refine Stone which increases the item option of <see cref="ItemOptionTypes.HarmonyOption"/>.
+/// Consume handler for the Higher Refine Stone which increases the item option of <see cref="ItemOptionTypes.HarmonyOption"/>.
 /// </summary>
 [Guid("A9F58DF6-06DB-4187-B386-9F00382333EE")]
 [PlugIn(nameof(HigherRefineStoneConsumeHandlerPlugIn), "Plugin which handles the higher refine stone consumption.")]
-public class HigherRefineStoneConsumeHandlerPlugIn : ItemUpgradeConsumeHandlerPlugIn
+public class HigherRefineStoneConsumeHandlerPlugIn : RefineStoneUpgradeConsumeHandlerPlugIn
 {
     /// <summary>
     /// Initializes a new instance of the <see cref="HigherRefineStoneConsumeHandlerPlugIn" /> class.
     /// </summary>
     public HigherRefineStoneConsumeHandlerPlugIn()
-        : base(new ItemUpgradeConfiguration(ItemOptionTypes.HarmonyOption, false, true, 0.5, ItemFailResult.SetOptionToLevelOne))
+        : base(new ItemUpgradeConfiguration(ItemOptionTypes.HarmonyOption, false, true, 0.8, ItemFailResult.SetOptionToBaseLevel))
```

Comment:
I'm pretty sure the original chance was 50 %. However, it's hard to find a reliable source.
Probably correct: "It is used to grant or upgrade +Bonus Harmony Option, with a 50% chance. It can also be used to remove +Jewel of Harmony Options from items." Source: https://muonlinefanz.com/tools/items/data/itemdb/High%20Refining%20Stone.php

Reply:
That is the page for S16 and above reference. Same source, but archive page (S15 and below):
https://muonlinefanz.com/archive/items/jewel-of-harmony-season-15/

Also, check the following references. Reference 1 is official GMO site S16 guide, which is confirmed by reference 2.
> - Updated Higher/Lower Refining Stones and JoH success rates ([reference 1](https://muonline.webzen.com/en/gameinfo/guide/detail/117), §3-6; [reference 2](https://github.com/kyleruss/emu-server/blob/202856a74c905c203b9b2795fd161f564ca8b257/Template/zGameServer/Data/CommonServer.cfg#L4))